### PR TITLE
Replace `core-maintainers` with `core-team`

### PR DESCRIPTION
**Pull request by luigigubello**
Try to solve #67

### CODE REVIEW

**Comment by luigigubello** [comment]
@eddie-knight almost ready, I need to update the documentation :)

**Comment by eddie-knight** [comment]
We should also make sure that any PR to modify the schema includes corresponding modifications to the spec. I can do that and push to this PR if that would help.

File: specification.md [hunk 1]
```diff
@@ -89,10 +94,25 @@ project-lifecycle:
 - `bug-fixes-only` (Required)
   - **Description:** Provide the maintenance status of the project by specifying if the maintainers fix only bugs without providing new features.
   - **Type:** Boolean.
-- `core-maintainers` (Conditionally required)
-  - **Description:** Provide the contacts of the project maintainers (emails, social profiles, websites, etc). This information can help consumers to contact the right people.
+- `core-maintainers`
+  - **Description:** [Deprecation notice] _`core-maintainers` will be deprecated in the next versions._ Provide the contacts of the project maintainers (emails, social profiles, websites, etc). This information can help consumers to contact the right people.
```
Comment (eddie-knight) [inline specification.md:98]:
```suggestion
  - **Description:** Provide the contacts of the project maintainers (emails, social profiles, websites, etc). This information can help consumers to contact the right people.
    - [Deprecation notice] _`core-maintainers` will be removed in v1.2.0_
```

**Comment by eddie-knight** [comment]
LGTM! @ me when you take this out of draft and I'm happy to approve

**Pull request merged**